`ChakraProvider.defaultProps` is implicitly typed any in generated declarations for `@chakra-ui/react`

### Description

`ChakraProvider.defaultProps` is currently declared as `var defaultProps`, which can cause TypeScript builds to fail in projects that depend on `@chakra-ui/react` and TypeScript strict mode.

### Link to Reproduction

https://stackblitz.com/edit/node-uqdb5a?file=tsconfig.json

### Steps to reproduce

1. Run `npm run build:tsc`
2. Observe the following output:
    ```
    $ tsc -p .
    node_modules/@chakra-ui/react/dist/index.d.ts:60:9 - error TS7005: Variable 'defaultProps' implicitly has an 'any' type.
     
    60     var defaultProps;
               ~~~~~~~~~~~~


    Found 1 error in node_modules/@chakra-ui/react/dist/index.d.ts:60
    ```


### Chakra UI Version

@chakra-ui/react 2.2.6

### Browser

N/A

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

This error appears when using TypeScript strict mode (see reproduction).

--- a/.changeset/three-jokes-swim.md
+++ b/.changeset/three-jokes-swim.md
@@ -2,4 +2,4 @@
 "@chakra-ui/react": minor
 ---
 
-Add explicit type declaration for `ChakraProvider.defaultProps`.
+Remove `ChakraProvider.defaultProps` and move logic inside the component.

--- a/.changeset/three-jokes-swim.md
+++ b/.changeset/three-jokes-swim.md
@@ -1,0 +1,5 @@
+---
+"@chakra-ui/react": minor
+---
+
+Add explicit type declaration for `ChakraProvider.defaultProps`.

--- a/packages/react/src/chakra-provider.tsx
+++ b/packages/react/src/chakra-provider.tsx
@@ -2,7 +2,7 @@ import {
   ChakraProvider as BaseChakraProvider,
   ChakraProviderProps as BaseChakraProviderProps,
 } from "@chakra-ui/provider"
-import { theme } from "@chakra-ui/theme"
+import { theme, Theme } from "@chakra-ui/theme"
 import { ToastProvider, ToastProviderProps } from "@chakra-ui/toast"
 
 export interface ChakraProviderProps extends BaseChakraProviderProps {
@@ -23,6 +23,10 @@ export function ChakraProvider({
       <ToastProvider {...toastOptions} />
     </BaseChakraProvider>
   )
+}
+
+export declare namespace ChakraProvider {
+  export let defaultProps: { theme: Theme }
 }
 
 ChakraProvider.defaultProps = {

--- a/packages/react/src/chakra-provider.tsx
+++ b/packages/react/src/chakra-provider.tsx
@@ -2,7 +2,7 @@ import {
   ChakraProvider as BaseChakraProvider,
   ChakraProviderProps as BaseChakraProviderProps,
 } from "@chakra-ui/provider"
-import { theme, Theme } from "@chakra-ui/theme"
+import { theme as defaultTheme } from "@chakra-ui/theme"
 import { ToastProvider, ToastProviderProps } from "@chakra-ui/toast"
 
 export interface ChakraProviderProps extends BaseChakraProviderProps {
@@ -17,18 +17,11 @@ export function ChakraProvider({
   toastOptions,
   ...restProps
 }: ChakraProviderProps) {
+  const theme = restProps.theme ?? defaultTheme
   return (
-    <BaseChakraProvider {...restProps}>
+    <BaseChakraProvider {...restProps} theme={theme}>
       {children}
       <ToastProvider {...toastOptions} />
     </BaseChakraProvider>
   )
-}
-
-export declare namespace ChakraProvider {
-  export let defaultProps: { theme: Theme }
-}
-
-ChakraProvider.defaultProps = {
-  theme,
 }

--- a/packages/react/src/chakra-provider.tsx
+++ b/packages/react/src/chakra-provider.tsx
@@ -14,12 +14,12 @@ export interface ChakraProviderProps extends BaseChakraProviderProps {
 
 export function ChakraProvider({
   children,
+  theme = defaultTheme,
   toastOptions,
   ...restProps
 }: ChakraProviderProps) {
-  const theme = restProps.theme ?? defaultTheme
   return (
-    <BaseChakraProvider {...restProps} theme={theme}>
+    <BaseChakraProvider theme={theme} {...restProps}>
       {children}
       <ToastProvider {...toastOptions} />
     </BaseChakraProvider>

